tests/components/build.rs

```diff
@@ -3,20 +3,70 @@ use std::env;
 use std::ffi::OsStr;
 use std::path::{Path, PathBuf};
 use std::process::{Output, Stdio};
+use std::sync::Arc;
 
-use anyhow::{bail, ensure, Context};
-use futures::try_join;
+use anyhow::{bail, ensure, Context, Result};
+use heck::ToKebabCase;
 use nkeys::KeyPair;
 use serde::Deserialize;
 use tokio::fs;
 use tokio::process::Command;
+use tokio::task::JoinSet;
 use wasmcloud_component_adapters::WASI_PREVIEW1_REACTOR_COMPONENT_ADAPTER;
 
+/// List of (manifest path, output artifact name) for all the packages used during test
+///
+/// Manifest paths should be relative to the directory containing this build.rs (i.e. tests/components)
+const WASI_WASM32_PACKAGES: [(&str, &str); 6] = [
+    ("./rust/Cargo.toml", "builtins-component-reactor"),
+    ("./rust/Cargo.toml", "interfaces-handler-reactor"),
+    ("./rust/Cargo.toml", "interfaces-reactor"),
+    ("./rust/Cargo.toml", "pinger-config-component"),
+    ("./rust/Cargo.toml", "ponger-config-component"),
+    (
+        "../../examples/rust/components/http-keyvalue-counter/Cargo.toml",
+        "http-keyvalue-counter",
+    ),
+];
+
+/// List of packages which should have output artifacts signed
+const WASI_WASM32_PACKAGES_SIGNED: [&str; 12] = [
+    "builtins-component-reactor",
+    "builtins-component-reactor-preview2",
+    "http-keyvalue-counter",
+    "http-keyvalue-counter-preview2",
+    "interfaces-handler-reactor",
+    "interfaces-handler-reactor-preview2",
+    "interfaces-reactor",
+    "interfaces-reactor-preview2",
+    "pinger-config-component",
+    "pinger-config-component-preview2",
+    "ponger-config-component",
+    "ponger-config-component-preview2",
+];
+
+/// Convert a hard-coded artifact name (see [`WASI_WASM32_PACKAGES`])
+///
+/// Pre-Rust 1.79, kebab cased project names would turn into kebab-cased artifacts
+#[rustversion::before(1.79)]
+fn generate_expected_artifact_name(project_name: &str) -> impl AsRef<str> {
+    project_name.to_kebab_case()
+}
+
+/// Convert a hard-coded artifact name (see [`WASI_WASM32_PACKAGES`])
+///
+/// Post-Rust 1.79, kebab cased project names turn into snake_case cased artifacts
+#[rustversion::since(1.79)]
+fn generate_expected_artifact_name(project_name: &str) -> impl AsRef<str> {
+    use heck::ToSnakeCase;
+    project_name.to_snake_case()
+}
```

Comment:
Oh boy, any changes that we need to address on the `wash build` side here?

Reply:
Nope I don't think so! `wash build` works just fine for me -- we don't really mess with the artifact names this directly/specifically.

Plus we also have this tested via CI so we'll know when this breaks